components/pageLoader.tsx

```diff
@@ -0,0 +1,114 @@
+import cachedData from "../cachedData.json";
+import Page from "./Page";
+import { JSObjectFromJSON } from "../src/dataProcessing";
+import { getJsonStructure } from "evergreen-org-crawler/src/index"
+import config from "evergreen-org-crawler/config.json"
+import { useEffect, useState } from "react";
+import CircularProgress from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+
```

Comment:
`pageLoader.tsx` should be `PageLoader.tsx` if it's a react component, if it is not it shouldn't be in the `components` folder.

Considering that this component mostly does data fetching, we can consider transforming it into a react hook to abstract things away a little bit.

_I added this comment in this blank line because I cannot post comment regarding an entire file._